app/api-v1/controllers/Recipe/index.js

```diff
@@ -1,45 +1,59 @@
 const { runProcess } = require('../../../utils/dscp-api')
-const { client } = require('../../../db')
+const db = require('../../../db')
 const { mapRecipeData } = require('./helpers')
 const { BadRequestError, NotFoundError } = require('../../../utils/errors')
 
-const throwErr = (type, path) => {
+const throwErr = (type, req) => {
   switch (type) {
     case 400:
       throw new BadRequestError({
         message: 'missing parameters',
-        path,
+        req,
       })
     case 404:
       throw new NotFoundError({
         message: 'not found',
-        path,
+        req,
       })
   }
 }
 
 module.exports = {
+  // TODO abstranct to transactions controller and first path e.g. /recipe /order is database model
+  // do this along with order
   transaction: {
+    getAll: async (req) => {
+      const { id } = req.params
+      if (!id) throwErr(400, req)
+
+      const transactions = await db.getAllRecipeTransactions(id)
```

Comment:
if I put an invalid `id` the response is `select * from "recipe_transactions" where "recipe_id" = $1 - invalid input syntax for type uuid:
"invalid-id"`, should we check and throw `400` with an invalid id?

Reply:
it does it already, database throws an error that gets picked up by err middleware